docs/adr/2023-06-19_024-persistence.md

```diff
@@ -21,35 +21,44 @@ Proposed
 Implement state persistence using [_Event Sourcing_](https://thinkbeforecoding.com/post/2013/07/28/Event-Sourcing-vs-Command-Sourcing). Practically, this means:
 
 1. Replace the `NewState` outcome with a `StateChanged` _event_ which can be part of the `Outcome` of `HeadLogic`'s `update` function, representing the _change_ to be applied to the current state.
-2. Add a dedicated [handle](/adr/4) to manage `StateChanged` events, applying them, and maintaining current `HeadState`
-3. Persist `StateChanged`s in an append-only log
-4. Upon node startup, reread `StateChanged` events log and reapply those to reset the `HeadState`
-3. _(Optional)_: Make `StateChanged` _invertible_
+2. Add an `aggregateState` function to manage applying `StateChanged` events on top of the current `HeadState` to keep it updated in-memory.
+3. Persist `StateChanged`s in an append-only log using a dedicated [handle](/adr/4).
+4. Upon node startup, reread `StateChanged` events log and reapply those to reset the `HeadState`.
+3. _(Optional)_: Make `StateChanged` _invertible_.
 
 The following sequence diagram illustrates new event handling in the `HeadLogic`:
 
 ```mermaid
 sequenceDiagram
-   EventQ ->> Node : nextEvent
-   Node ->> State : getState
-   activate State
-   State -->> Node : state: HeadState
-   Node ->> HeadLogic: update(state)
-   activate HeadLogic
-   HeadLogic -->> Node : [st:StateChanged, ot:OtherEffect]
-   Node ->> State: setState(st)
-   activate State
-   State ->> State: persist(st)
-   State -->> Node: ok
-   Node -) Other: handle(ot)
+   Node ->> EventQ : nextEvent
+   activate EventQ
+   EventQ -->> Node : event
+   deactivate EventQ
+   critical modifyHeadState
+    Node ->> State : getState
```

Comment:
There is not really a `getState` and `setState` function. Instead of the critical area, I think we could use a `modifyHeadState` call/activation box (using `+` in the mermaid arrow)